state/storage.go

```diff
@@ -388,12 +388,32 @@ func validateStorageConstraints(st *State, cons map[string]StorageConstraints, c
 				charmMeta.Name, name,
 			)
 		}
-		if cons.Pool != "" {
-			conf, err := st.EnvironConfig()
-			if err != nil {
-				return errors.Trace(err)
+		conf, err := st.EnvironConfig()
```

Comment:
This looks fine for now, though I think we should rework this later so that state is not calling into the storage package. It would be better if we made these sorts of decisions in the API server code. Validation is tricky, and probably means extending the policy code.